data/settings/src/main/proto/com/google/jetpackcamera/settings/jca_settings.proto

```diff
@@ -44,5 +44,5 @@ message JcaSettings {
   bool stabilize_video_supported = 11;
   bool stabilize_preview_supported = 12;
   TargetFrameRate target_frame_rate = 13;
-
+  int32 max_frame_rate = 14;
```

Comment:
The max frame rate doesn't necessarily mean that every frame rate underneath it is supported. We should check that [15, 15], [30, 30] and [60, 60] are supported explicitly, and add them to this proto as a list. So this field would change to:

`repeated int32 supported_frame_rates = 14;`

Reply:
I think this raises a question on what exactly the user may expect the buttons on the settings to convey and do. 
I did my current implementation with this in mind: the device has some fps ranges that reach up to 15/30/60. Selecting one of the options will set the device to find a frame rate that best matches it.

I think limiting the options to only reflect an exact match could potentially disable choices on the setting, despite ranges existing that contain the desired frame rate.
Rather than check for an exact match, what do you think about searching for a range that actually **contains** 15/30/60 instead of the current implementation or searching for an exact match?

Reply:
Video frame rates historically are fixed frame rates. This means [15, 15], [30, 30], or [60, 60] (other standard frame rates are [24, 24] and [25, 25]). These will commonly also be supported frame rates listed in device capabilities. If the user selects 60fps, and we set something like [15, 60], and they're in a dark room, then the AE algorithm may actually choose something closer to 15fps. This will seem like broken behavior to the user who is expecting a 60fps video.

If we want to allow variable frame rates, we need to explicitly call it out as something like "Auto" frame rate or "Variable" frame rate. But there isn't a consistent definition of "variable" frame rate across all devices, so for a consistent behavior we'll need to either add more options for the user to select or explicitly call out frame rate ranges.

Let's stick with the standard fixed frame rates for now. If there is a request for a variable frame rate setting, we can think about how to implement it later.